tools/src/main/java/com/graphhopper/tools/Measurement.java

```diff
@@ -781,10 +784,7 @@ public int doCalc(boolean warmup, int run) {
                 if (querySettings.alternative)
                     req.setAlgorithm(ALT_ROUTE);
 
-                if (querySettings.withInstructions)
-                    req.setPathDetails(Arrays.asList(Parameters.Details.AVERAGE_SPEED));
```

Comment:
This indeed looks strange. We are enabling instructions by enabling path details?

Reply:
We enabled two things, so we could have also renamed withInstructions into withInstructionsAndPD (?)